apps/worker/src/app/workflow/usecases/send-message/send-message-email.usecase.ts

```diff
@@ -141,22 +145,28 @@ export class SendMessageEmail extends SendMessageBase {
       this.sendSelectedIntegrationExecution(command.job, integration),
     ]);
 
+    const template = await this.processVariants(command, tenant, subscriber, command.payload);
+
+    if (template) {
+      step.template = template;
+    }
+
     const overrides: Record<string, any> = Object.assign(
       {},
       command.overrides.email || {},
       command.overrides[integration?.providerId] || {}
     );
 
     let html;
-    let subject = '';
+    let subject = step?.template?.subject || '';
     let content;
 
     const payload = {
-      subject: emailChannel.template.subject || '',
-      preheader: emailChannel.template.preheader,
-      content: emailChannel.template.content,
-      layoutId: overrideLayoutId ?? emailChannel.template._layoutId,
-      contentType: emailChannel.template.contentType ? emailChannel.template.contentType : 'editor',
+      subject,
```

Comment:
Looks like we were saving an empty string as a subject, not sure if there was a reason for that.